source/app/viewoptions.h

```diff
@@ -19,20 +19,34 @@
 #define APP_VIEWOPTIONS_H
 
 #include <optional>
+#include <array>
+#include <algorithm>
 
 /// Stores any view options that are not saved with the score (e.g. the current
 /// zoom level or the active score filter).
 class ViewOptions
 {
+public:
+    static constexpr double MIN_ZOOM = 25;
+    static constexpr double MAX_ZOOM = 300;
+    static constexpr std::array<unsigned short, 13> ZOOM_LEVELS = { 25, 50, 70, 80, 90, 100, 110, 125, 150, 175, 200, 250, 300 };
+
 public:
     ViewOptions();
 
     const std::optional<int> &getFilter() const { return myFilter; }
     void setFilter(int filter) { myFilter = filter; }
     void clearFilter() { myFilter.reset(); }
 
-    double getZoom() const { return myZoom; }
-    void setZoom(double percent) { myZoom = percent; }
+    int getZoom() const { return myZoom; }
```

Comment:
@cameronwhite was there a particular reason for the zoom to be a double? I believe a `short` would suit it better. What do you think? (I know I've changed it to `int` here. A simple way to avoid a settings converter)

Reply:
I don't think I had any particular reason for double - changing to `int` would be fine!
I usually avoid lower capacity types like `short` unless there's a real need for it.